File: ccdr/makeitso/iOS/MakeItSo/MakeItSo/Models/Task.swift
import Foundation
import FirebaseFirestore

struct Task: Codable, Identifiable {
  @DocumentID var id: String?
  var title: String
  var isCompleted: Bool
  var priority: TaskPriority
  var dueDate: Date? // Optional, as tasks might not have a deadline
  @ServerTimestamp var createdAt: Date?
  var userId: String?
}

enum TaskPriority: String, Codable, CaseIterable {
  case low = "Low"
  case medium = "Medium"
  case high = "High"
}

File: ccdr/makeitso/Android/MakeItSo/app/src/main/java/com/google/firebase/example/makeitso/data/model/Task.kt
package com.google.firebase.example.makeitso.data.model

import com.google.firebase.firestore.DocumentId
import com.google.firebase.firestore.ServerTimestamp
import java.util.Date

enum class TaskPriority(val value: String) {
    LOW("Low"),
    MEDIUM("Medium"),
    HIGH("High");

    override fun toString(): String = value
}

data class Task(
    @DocumentId val id: String = "",
    val title: String = "",
    val description: String = "",
    val isCompleted: Boolean = false,
    val priority: TaskPriority = TaskPriority.LOW,
    val dueDate: Date? = null,
    @ServerTimestamp val createdAt: Date? = null,
    val userId: String? = null
)
